cpp/core/shuffle/Utils.h

```diff
@@ -72,4 +72,34 @@ arrow::Result<std::shared_ptr<arrow::RecordBatch>> makeUncompressedRecordBatch(
 
 std::shared_ptr<arrow::Buffer> zeroLengthNullBuffer();
 
+class MmapFileStream : public arrow::io::InputStream {
```

Comment:
Could you please add some comments to explain the usage/functionality for this class?